test/cpp/jit/test_interpreter.cpp

```diff
@@ -5,27 +5,71 @@ namespace torch {
 namespace jit {
 
 void testInterp() {
-  constexpr int batch_size = 4;
-  constexpr int input_size = 256;
-  constexpr int seq_len = 32;
+  {
+    auto graph = std::make_shared<Graph>();
+    std::unordered_map<std::string, Value*> vmap;
+    parseIR(
+        R"IR(
+graph(%a.1 : Tensor,
+      %b.1 : Tensor):
+  %t : Float(2:2, 2:1), %t0 : Float(2:2, 2:1), %type_matched : bool = prim::TypeCheck(%a.1, %b.1)
```

Comment:
We should have more tests to cover following scenarios:
* 0 inputs/outputs
* number of inputs and outputs mismatch
* typecheck is expected to return true
* typecheck is expected to return false
* expected sizes differ from each other for different outputs
* we check for properties like 'requires_grad' or 'device'

Reply:
PS: If it's too cumbersome to check for invalid inputs in the tests, it's fine not to include such tests - let's just make sure that if it ever happens we fail in a clearly understandable way, with an assertion telling us what's going on.